extension/src/createProject/SpecifySourcePackageNameStep.ts

```diff
@@ -39,15 +45,21 @@ export class SpecifySourcePackageNameStep implements IProjectCreationStep {
                 );
             }
             disposables.push(
+                inputBox.onDidChangeValue(async () => {
+                    const normalizedName = await getNormalizedPackageNameTrigger(inputBox.value);
+                    if (normalizedName !== inputBox.value) {
+                        inputBox.enabled = false;
+                        inputBox.validationMessage = `Invalid source package name, suggest name: ${normalizedName}`;
+                    } else {
+                        inputBox.enabled = true;
+                        inputBox.validationMessage = undefined;
+                    }
+                }),
                 inputBox.onDidAccept(async () => {
-                    const normalizedName = await metadata.client.getNormalizedPackageName(inputBox.value);
-                    if (normalizedName === inputBox.value) {
+                    if (inputBox.enabled) {
```

Comment:
you should not remove validation here. Otherwise, if I type a `-` very quickly and press `enter`, the value containing `-` will be accepted.

Reply:
Right. validation has been added in https://github.com/microsoft/vscode-gradle/pull/1213/commits/b3547eb4037364efb1e3264200f1e769b33b84d4